tests/ut_recentcontactsmodel/recentcontactsmodeltest.cpp

```diff
@@ -38,33 +39,38 @@ static void addEvents(int from, int to)
     // by adjusting the start times for events with duration (CallEvent)
     // so that their end times are all in the expected order.
 
+    // Wait briefly after adding IM and SMS events, otherwise the additions are not seen by the model.
```

Comment:
Hmm.  We cannot use a signal spy for this instead of waits?  Also, why 100 msec, surely 1 msec is enough, if the signal emission itself is synchronous?

Reply:
That really would be ideal, but I haven't been able to figure out what is being done in the background that is making the model unable to load these events more immediately. 1 msec is not enough, there's more happening beyond the next event loop.